src/cli/lint/internalLints/upgrade/patches/8.2.1/patchDockerCompose.ts

```diff
@@ -0,0 +1,77 @@
+import { inspect } from 'util';
+
+import fg from 'fast-glob';
+import { readFile, writeFile } from 'fs-extra';
+
+import type { PatchFunction, PatchReturnType } from '../..';
+import { log } from '../../../../../../utils/logging';
+
+const DOCKER_COMPOSE_VERSION_REGEX = /^version: ['"]?\d+(\.\d+)*['"]?\n*/m;
+
+const fetchFiles = async (files: string[]) =>
+  Promise.all(
+    files.map(async (file) => {
+      const contents = await readFile(file, 'utf8');
+
+      return {
+        file,
+        contents,
+      };
+    }),
+  );
+
+const patchDockerComposeFiles: PatchFunction = async ({
+  mode,
+}): Promise<PatchReturnType> => {
+  const maybeDockerComposeFiles = await Promise.resolve(
+    fg(['docker-compose*.yml']),
+  );
+
+  if (!maybeDockerComposeFiles.length) {
+    return {
+      result: 'skip',
+      reason: 'no docker-compose files found',
+    };
+  }
+
+  const dockerComposeFiles = await fetchFiles(maybeDockerComposeFiles);
+
+  const dockerComposeFilesToPatch = dockerComposeFiles.filter(({ contents }) =>
```

Comment:
I think if we wanted to be rigorous about this, we'd do the .replace up here (or .match with the regex), and then check if any are different; technically you could have version: but not match the regex below.